### PR TITLE
Add build step for ICPC (with fixes)

### PR DESCRIPTION
**Pull request by falbrechtskirchinger**
Fix the remaining issues of PR #3229.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1115,6 +1103,19 @@ The following compilers are currently used in continuous integration at [AppVeyo
 | Clang 13.0.1 (13.0.1-++20211015123032+cf15ccdeb6d5-1exp120211015003613.5)                              | Ubuntu 20.04.3 LTS | GitHub Actions |
 | Clang 14.0.0 (14.0.0-++20211221052852+55c71c9eac9b-1exp120211221172954.95)                             | Ubuntu 20.04.3 LTS | GitHub Actions |
 | Clang 15.0.0 (15.0.0-++20220403052648+896770c9a92e-1~exp1~20220403172744.209)                          | Ubuntu 20.04.3 LTS | GitHub Actions |
+| GCC 4.8.5 (Ubuntu 4.8.5-4ubuntu2)                                                                      | Ubuntu 20.04.3 LTS | GitHub Actions |
```
Comment (nlohmann) [inline README.md:1106]: Sorted alphabetically?

File: tests/src/unit-wstring.cpp [hunk 2]
```diff
@@ -32,6 +32,7 @@ SOFTWARE.
 #include <nlohmann/json.hpp>
 using nlohmann::json;
 
+#ifndef __INTEL_COMPILER
```
Comment (nlohmann) [inline tests/src/unit-wstring.cpp:35]: Please add a comment why this block is excluded.

File: tests/src/unit-regression2.cpp [hunk 3]
```diff
@@ -512,12 +512,14 @@ TEST_CASE("regression tests 2")
 
     SECTION("issue #1647 - compile error when deserializing enum if both non-default from_json and non-member operator== exists for other type")
     {
+#if !(defined(__INTEL_COMPILER) && __cplusplus >= 202000)
```
Comment (nlohmann) [inline tests/src/unit-regression2.cpp:515]: Please add a comment why this block is excluded.

File: README.md (hunk 1 above)
Comment (falbrechtskirchinger) [reply README.md:1106]: Yes. Was there a reason why it wasn't before?

Comment (nlohmann) [reply README.md:1106]: Maybe for historic reasons. 🤷‍♂️

File: tests/src/unit-regression2.cpp [hunk 4]
```diff
@@ -793,8 +796,8 @@ TEST_CASE("regression tests 2")
         const auto j_path = j.get<nlohmann::detail::std_fs::path>();
         CHECK(j_path == text_path);
 
-#if !defined(_MSC_VER) && !(defined(__GNUC__) && __GNUC__ == 8 && __GNUC_MINOR__ < 4)
-        // works everywhere but on MSVC and GCC <8.4
+#if defined(__clang__) || ((defined(__GNUC__) && !defined(__INTEL_COMPILER)) && (__GNUC__ > 8 || (__GNUC__ == 8 && __GNUC_MINOR__ >= 4)))
```
Comment (nlohmann) [inline tests/src/unit-regression2.cpp:799]: Something is not properly formatted: The CI complains:

```
Formatted  /__w/json/json/tests/src/unit-regression2.cpp
```

**Review by nlohmann — APPROVED** [review]
Looks good to me.

**Comment by nlohmann** [comment]
Thanks!

**Pull request merged**